taichi/ir/type_factory.cpp

```diff
@@ -2,6 +2,11 @@
 
 TLANG_NAMESPACE_BEGIN
 
+TypeFactory &TypeFactory::get_instance() {
+  static TypeFactory type_factory;
+  return type_factory;
+}
```

Comment:
We probably want to stick to the old implementation in `Program::get_type_factory` since if `type_factory` get destroyed before other global variables that use it, the program won't finish smoothly.

Reply:
Btw, since #1962 is in, feel free to remove `Program::get_type_factory()` in a different PR :-) Thanks!

Reply:
> We probably want to stick to the old implementation in `Program::get_type_factory` since if `type_factory` get destroyed before other global variables that use it, the program won't finish smoothly.

Yes, you are right. let me change it to the old implementation.